src/Libraries/Microsoft.Extensions.Http.Resilience/Internal/RequestMessageSnapshot.cs

```diff
@@ -64,24 +86,76 @@ bool IResettable.TryReset()
         _method = null;
         _version = null;
         _requestUri = null;
+        if (_content is StreamContent)
+        {
+            // a snapshot's StreamContent is always a unique copy (deep clone)
+            // therefore, it is safe to dispose when snapshot is no longer needed
+            _content.Dispose();
+        }
+
         _content = null;
 
         return true;
     }
 
     void IDisposable.Dispose() => _snapshots.Return(this);
 
-    private void Initialize(HttpRequestMessage request)
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Resilience", "EA0014:The async method doesn't support cancellation", Justification = "Past the point of no cancellation.")]
+    private static async Task<(HttpContent? content, HttpContent? clonedContent)> CloneContentAsync(HttpContent? content)
     {
-        if (request.Content is StreamContent)
+        HttpContent? clonedContent = null;
+        if (content is not null)
```

Comment:
This introduces performance regression for non-streamed content. Create an extra branch for `StreamContent` where this logic belongs.

Reply:
@martintmk can you expand on why you believe this will cause a performance regression?

The cloning of the `HttpContent` (i.e. this method) only executes if the `request.Content` is `StreamContent`. See the other places in this class where this `CloneContentAsync` method is called for more context. You will see that each call is wrapped in a `if (request.Content is StreamContent)` check. I did this specifically to prevent any performance regression for existing non-Streamed content users of the `AddStandardHedgingHandler` API.

Also, if you are referring to the `if (request.Content is StreamContent)` check as being the culprit of a new performance regression, please observe that this check already existed before my change. The only difference is that, currently, if `request.Content is StreamContent` is true, then the `RequestMessageSnapshot` methods throw an `InvalidOperationException`. Therefore, perf should remain the same for existing, non-Streamed content users.

Reply:
I see, thanks for detailed explanation. Can you still try to avoid the extra branches by condensing the code into a single `PrepareContentAsync` method
It would also reduce the number of mutants you have to kill.

Reply:
Ahh I see what you mean. I understand where you are coming from; however, I don't think encapsulating the content operations behind a single method will achieve the additional clarity or eliminate the required mutation tests as you are suggesting:
1. The mutation tests still need to be accounted for regardless.
2. I don't believe it would add any additional clarity. In fact, I believe it might do the opposite and here's why: the method would need to operate in both directions (i.e. cloning a passed-in request object into the given `RequestMessageSnapshot` instance fields and cloning the given `RequestMessageSnapshot` instance into a new request object). I don't see any clear way to represent that behind a single method without it creating more confusion. It even seems like it would be a [code smell](https://en.wikipedia.org/wiki/Code_smell).